spec/active_record/monetizable_spec.rb

```diff
@@ -854,6 +854,39 @@ class SubProduct < Product
           expect(product.read_monetized(:price, :price_cents).to_s).to eq('14,0')
         end
       end
+
+      context "with a monetized attribute that is nil" do
+        let(:service) do
+          Service.create(discount_cents: nil)
+        end
+        let(:default_currency_lambda) { double("Default Currency Fallback") }
+        subject { service.read_monetized(:discount, :discount_cents, options) }
+
+        around(:each) do |example|
+          service # Instantiate instance which relies on Money.default_currency
+          original_default_currency = Money.default_currency
+          Money.default_currency = -> { default_currency_lambda.read_currency }
+          example.run
+          Money.default_currency = original_default_currency
+        end
+
+        context "when allow_nil options is not set" do
```

Comment:
I think you can remove this test case since if the `allow_nil` is not set it should follow the regular path and is probably tested somewhere else